microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java

```diff
@@ -0,0 +1,81 @@
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Class aggregating all subsystem related tests. If there ever will be configuration options for subsystem, this is the
+ * place to add tests for them.
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class MicroProfileJwtSubsystemTestCase {
+
+    private static final String MP_JWT_SUBSYSTEM_NAME = "microprofile-jwt-smallrye";
+
+    private static final Address MP_JWT_SUBSYSTEM_ADDRESS = Address.subsystem(MP_JWT_SUBSYSTEM_NAME);
+
+    private OnlineManagementClient client;
+    private boolean wasSubsystemRemovedInPrepare = false;
+
+    @Before
+    public void before() throws ConfigurationException, IOException, OperationException, TimeoutException,
+            InterruptedException {
+        client = ManagementClientProvider.onlineStandalone();
+        final Operations ops = new Operations(client);
+
+        if (ops.exists(MP_JWT_SUBSYSTEM_ADDRESS)) {
+            ops.remove(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+            new Administration(client).reloadIfRequired();
+            wasSubsystemRemovedInPrepare = true;
+        }
+    }
+
+    @Before
+    public void after() throws IOException, TimeoutException, InterruptedException {
+        final Operations ops = new Operations(client);
+
+        if (wasSubsystemRemovedInPrepare) {
+            ops.add(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+            new Administration(client).reloadIfRequired();
+        }
+
+        if (client != null) {
+            client.close();
+        }
+    }
+
+    /**
+     * @tpTestDetails Add and remove MicroProfile JWT subsystem.
+     * @tpPassCrit Subsystem is successfully added and removed.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void addRemoveSubsystemTest() throws IOException, OperationException, TimeoutException,
+            InterruptedException {
+        final Operations ops = new Operations(client);
+        final Administration administration = new Administration(client);
+
+        ops.add(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+        administration.reloadIfRequired();
+        Assert.assertTrue(ops.exists(MP_JWT_SUBSYSTEM_ADDRESS));
+        ops.remove(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+        administration.reloadIfRequired();
+        Assert.assertFalse(ops.exists(MP_JWT_SUBSYSTEM_ADDRESS));
+    }
+
```

Comment:
What about calling wasSubsystemRemovedInPrepare twice ? Second call should fail + error message should be reasonable.

Reply:
I'm sorry, I'm not sure I follow - are you pointing out that I should try to remove the subsystem for second time and verify, the operation fails?

Reply:
yes, new test case candidate.

Reply:
this can be new PR ...